flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesFactory.java

```diff
@@ -63,9 +63,21 @@ public static BigQueryServicesFactory instance(BigQueryConnectOptions options) {
      */
     public BigQueryServices.StorageReadClient storageRead() throws IOException {
         if (isTestingEnabled) {
```

Comment:
instead of using this isTestingEnabled, can we use dependency inject for the "services" object? have the BigQueryServicesFactory take in a services object in the constructor that can be mocked or faked for testing

Reply:
This is a non-trivial change which will require significant code change. Created buganizer task `b/330782741` for internal tracking.

Reply:
cool that works for me. you can resolve this convo